### PR TITLE
ReactiveObjC: upgrade deployment target to 12.0 and eliminate Xcode12 warnings.

### PR DESCRIPTION
**Pull request by arangato**
Here is a brief summary:

- Replace deprecated OSAtomic APIs with stdatomic.
- Use latest Nimble and Quick libraries.
- Raise deployment target to 12.0
- Enable recommended warning for quoited framework includes
    - Disabled it for tests because there are several of these in Nimble and Quick
- Use Xcode 12 in circle CI.

See commit messages for more details.

### CODE REVIEW

**Comment by arangato** [comment]
@barakwei this one is ready for review.

File: ReactiveObjC/RACCommand.m [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 @interface RACCommand () {
   // Atomic backing variable for `allowsConcurrentExecution`.
-  volatile uint32_t _allowsConcurrentExecution;
+  atomic_uint _allowsConcurrentExecution;
```
[the remark this answers is not in the document]
Comment (arangato) [reply ReactiveObjC/RACCommand.m:29]: The usage seems sound. I prefer diverging as little as possible. Cherrypicks are complicated enough already.

[the remark this answers is not in the document]
Comment (StatusReport) [reply ReactiveObjC/RACCommand.m:29]: So why not `uint32_t`? Why do you assume that the merged code is better? No one really maintains that repo...

**Comment by arangato** [comment]
Tests with TSan pass as well. No issues spotted. 
There is a [PR#179.](https://github.com/ReactiveCocoa/ReactiveObjC/pull/179) upstream that replaces #178, but it's the same thing squashed.

File: ReactiveObjC/RACCommand.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (arangato) [reply ReactiveObjC/RACCommand.m]: Sorry. It got mixed up in my head with `RACDynamicSequence`.

File: ReactiveObjC/RACCommand.m [hunk 2]
```diff
@@ -55,9 +55,9 @@ - (BOOL)allowsConcurrentExecution {
 
 - (void)setAllowsConcurrentExecution:(BOOL)allowed {
   if (allowed) {
-    OSAtomicOr32Barrier(1, &_allowsConcurrentExecution);
+    atomic_fetch_or(&_allowsConcurrentExecution, 1);
```
Comment (barakwei) [inline ReactiveObjC/RACCommand.m:58]: if it's atomic bool, use true and false.

**Comment by arangato** [comment]
done.

**Review by barakwei — APPROVED** [review]

**Pull request merged**